controller/state/statedb_dml.go

```diff
@@ -97,4 +97,15 @@ const (
 	queryHeadSQL = `
 		SELECT cid FROM record_updates WHERE status = $1 AND worked_by = $2
 	`
+
+	workerTasksByStatusSQL = `
+	SELECT tasks.uuid, tasks.data FROM tasks
+	INNER JOIN task_status_ledger ON tasks.uuid=task_status_ledger.uuid
+	WHERE tasks.worked_by = $1 AND task_status_ledger.status = $2
+`
+
+	unassignTaskSQL = `
+		UPDATE tasks SET data = $2, worked_by = NULL, cid = $3
```

Comment:
should we do this, or should we make a new initial state task? having a log that jumps back to unassigned after getting partially executed may be harder to infer anything about when we look for source of failure at the end.

Reply:
I'm not entirely sure why I feel this way, but I feel that preserving the UUID will be relevant. Note this does actually wipe the logs that the reputation service will see.